### PR TITLE
ICU-21182 Adding UNumberRangeFormatter.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21182
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
Squashed, and I also caught and fixed that the commit was misticketted!

**Comment by sffc** [comment]
Can I have an approval for the squashed commit, please? :smiley:

File: icu4c/source/i18n/unicode/unumberrangeformatter.h [hunk 1]
```diff
@@ -0,0 +1,431 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+#ifndef __UNUMBERRANGEFORMATTER_H__
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/unumberrangeformatter.h:7]: The guard should be between the copyright and the first include, like in other files. Remember to move the corresponding endif+comment to the last line, too.

Comment (sffc) [reply icu4c/source/i18n/unicode/unumberrangeformatter.h:7]: Seems that we're not consistent here (I spot several files with my order), but fixed via commit amendment.

**Pull request merged**